### ISSUE TITLE
[rancher-dns] adding a scheduling rule breaks service

### ISSUE THREAD

**Issue by ebr** [rancher/rancher#2834]
Steps to reproduce:
  - create a route53 stack/service from Catalog; confirm working
  - upgrade the service, adding a scheduling rule to run only on hosts with a specific label

Result:
  - new container is scheduled on the new host, as expected;
  - container keeps restarting but unable to fully initialize
  - In container's logs:
   
```
Configured Route53 with hosted zone \"my.zone.\" in region \"us-east-1\" 
Processing event: &docker.APIEvents{Status:\"start\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"-simulated-\", Time:0}
CATTLE_URL is not set
Processing event: &docker.APIEvents{Status:\"die\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565490}
Processing event: &docker.APIEvents{Status:\"die\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565476}
Processing event: &docker.APIEvents{Status:\"start\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565456}
Container locked. Can't run StartHandler. ID: [386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457]
Starting Rancher External DNS service
Configured Route53 with hosted zone \"my.zone.\" in region \"us-east-1\" 
Starting Rancher External DNS service
Processing event: &docker.APIEvents{Status:\"die\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565483}
Assigning IP [re.da.ct.ed/16], ContainerId [386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457], Pid [3657]
Configured Route53 with hosted zone \"my.zone.\" in region \"us-east-1\" 
Starting Rancher External DNS service
Processing event: &docker.APIEvents{Status:\"start\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565488}
Starting Rancher External DNS service
Processing event: &docker.APIEvents{Status:\"start\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565492}
Container locked. Can't run StartHandler. ID: [386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457]
Starting Rancher External DNS service
Processing event: &docker.APIEvents{Status:\"die\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565493}
Processing event: &docker.APIEvents{Status:\"start\", ID:\"386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457\", From:\"rancher/external-dns:v0.1.7\", Time:1448565412}
Assigning IP [re.da.ct.ed/16], ContainerId [386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457], Pid [13233]
Assigning IP [re.da.ct.ed/16], ContainerId [386b88ab4c0fb03864a5a3cf7211cd9641a34c07a47bd59d2ef5ea8c84e03457], Pid [20721]
Starting Rancher External DNS service

...etc
```    

the only way i've found to recover is to delete the stack and re-create.

**Comment by vincent99** [comment]
I believe this is because of #2342, some of the labels needed aren't copied.

**Comment by deniseschannon** [comment]
Yes, this is definitely because none of the system labels are copied over for upgrade. This includes any labels that Rancher created during service creation. (e.g. io.rancher.stack.name

Easy steps to Reproduce and see issue:
1. Launch route53 service.
2. Click on Labels tab in route53 detailed service view.
3. Upgrade route53 service. Click on Labels tab in detailed service

Result: None of the previous labels are copied over during upgrade.

Expected Result: All labels to be copied over.

**Comment by vincent99** [comment]
@deniseschannon This is supposed to work now with the changes to how labels are copied in the UI

**Comment by deniseschannon** [comment]
Verified this is fixed in v0.58.0-rc2
